pkg/domain/stateDiffConfig.go

```diff
@@ -0,0 +1,43 @@
+package domain
+
+type DoguConfigDiff struct {
+	SensibleDoguConfigDiff SensibleDoguConfigDiff
+	NormalDoguConfigDiff   NormalDoguConfigDiff
+}
+
+type SensibleDoguConfigDiff []ConfigKeyDiff
+
+type NormalDoguConfigDiff []ConfigKeyDiff
+
+type ConfigKeyDiff struct {
+	Key      string
+	Actual   ConfigValue
+	Expected ConfigValue
+	Action   ConfigAction
+}
+
+type ConfigValue struct {
+	Value  string
+	Exists bool
+}
+
+type GlobalConfigDiff []ConfigKeyDiff
+
+type ConfigAction string
+
+const (
+	ConfigActionNone   ConfigAction = "none"
+	ConfigActionSet    ConfigAction = "set"
+	ConfigActionRemove ConfigAction = "remove"
+)
+
+func getNeededConfigAction(expected ConfigValue, actual ConfigValue) ConfigAction {
```

Comment:
🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>func `getNeededConfigAction` is unused

Reply:
yes, we wrote this code to show how to check the configDiff.